src/qibo/backends/__init__.py

```diff
@@ -11,43 +11,34 @@ def __init__(self):
 
         # check if numpy is installed
         if self.check_availability("numpy"):
-            from qibo.backends.numpy import NumpyDefaultEinsumBackend, NumpyMatmulEinsumBackend
-            self.available_backends["numpy"] = NumpyDefaultEinsumBackend
-            self.available_backends["numpy_defaulteinsum"] = NumpyDefaultEinsumBackend
-            self.available_backends["numpy_matmuleinsum"] = NumpyMatmulEinsumBackend
+            from qibo.backends.numpy import NumpyBackend
+            self.available_backends["numpy"] = NumpyBackend
         else:  # pragma: no cover
             raise_error(ModuleNotFoundError, "Numpy is not installed.")
 
         # check if tensorflow is installed and use it as default backend.
         if self.check_availability("tensorflow"):
-            from qibo.backends.tensorflow import TensorflowDefaultEinsumBackend, TensorflowMatmulEinsumBackend
             os.environ["TF_CPP_MIN_LOG_LEVEL"] = str(config.LOG_LEVEL)
-            import tensorflow as tf  # pylint: disable=E0401
-            self.available_backends["defaulteinsum"] = TensorflowDefaultEinsumBackend
-            self.available_backends["matmuleinsum"] = TensorflowMatmulEinsumBackend
-            self.available_backends["tensorflow_defaulteinsum"] = TensorflowDefaultEinsumBackend
-            self.available_backends["tensorflow_matmuleinsum"] = TensorflowMatmulEinsumBackend
+            import tensorflow as tf # pylint: disable=E0401
+            from qibo.backends.tensorflow import TensorflowBackend
+            self.available_backends["tensorflow"] = TensorflowBackend
+            active_backend = "tensorflow"
             if self.check_availability("qibotf"):
                 from qibo.backends.tensorflow import TensorflowCustomBackend
-                self.available_backends["custom"] = TensorflowCustomBackend
-                self.available_backends["tensorflow"] = TensorflowCustomBackend
+                self.available_backends["qibotf"] = TensorflowCustomBackend
+                active_backend = "qibotf"
             else: # pragma: no cover
                 log.warning("Einsum will be used to apply gates with Tensorflow. "
                             "Removing custom operators from available backends.")
-                self.available_backends["tensorflow"] = TensorflowDefaultEinsumBackend
-            active_backend = "tensorflow"
         else:  # pragma: no cover
             # case not tested because CI has tf installed
             log.warning("Tensorflow is not installed. Falling back to numpy. "
                         "Numpy does not support Qibo custom operators and GPU. "
                         "Einsum will be used to apply gates on CPU.")
-            # use numpy for defaulteinsum and matmuleinsum backends
```

Comment:
Looks good. Let me suggest to update the warning messages with actions that should be taken from the user in order to enable the backend, e.g. where we could say "in order to activate the tensorflow backend please do `pip install tensorflow`, same for `qibotf` and `numpy`.

Reply:
Thanks for the review. I updated the warnings, please have a second look and let me know if you agree.